tests/neg/i6606.scala

```diff
@@ -0,0 +1,8 @@
+val t0: Int = 0
+val t1: scala.compiletime.S[-1] = 0 // error
+val t2: scala.compiletime.S[Int] = 1
```

Comment:
It is not clear to me why this line type checks.

Reply:
I guess it is because type `1 =:= S[0] <:< S[Int]`

Reply:
We need to figure out the precise spec for `S`, whether it's co-variant, contra-variant, non-variant.